Libraries/dotNetRDF/Writing/Formatting/Rfc3987Formatter.cs

```diff
@@ -33,57 +33,97 @@ internal static class Rfc3987Formatter
         public static string EscapeUriString(string uriString)
         {
             var builder = new StringBuilder();
-            foreach (var c in uriString)
+            using (var enumerator = uriString.GetEnumerator())
```

Comment:
By changing to enumerator I can peek the two chars following a `%` and append verbatim. The only reason to do otherwise is when the `%` is not followed by a valid escape sequence, dangling at the end of the string